Backend/src/main/java/ca/mcgill/cooperator/service/ReportSectionService.java

```diff
@@ -127,11 +130,11 @@ public ReportSection updateReportSection(
             rs.setEmployerReport(er);
             reportSectionRepository.save(rs);
             boolean contains = false;
-            List<ReportSection> sections = er.getReportSections();
+            Set<ReportSection> sections = er.getReportSections();
             for (ReportSection section : sections) {
                 if (section.getId() == rs.getId()) {
-                    int index = sections.indexOf(section);
```

Comment:
@susanmatuszewski what is this doing? Isn't this redundant since it sets the index to the same report section that's already there?

Reply:
@kraglalbert i asked susan and i guess its to update the list of the parent that this section belongs to or something like that

Reply:
okay so yea what emma said but its possible the cascades take care of it

Reply:
okay sorry for being late with this, the point of this loop is basically check to see if the report section thats being added already exists with the employer report thats being added to it. for ex is the employer report were to change for this report section, a new one would need to be added to the employer reports' set of report sections. if the employer report stays the same then the report section needs to be updated with the newly updated report section. The << section.add(section); >> should instead be << section.add(rs); >> since rs is the updated report section. It's also possible cascades take care of this but i think i just dont really trust those since they deletes dont seem to work consistently. just message me if you have any questions about this its kinda confusing

Reply:
@susanmatuszewski yes i added that the other day and it passed so i think its fine with the loop in the end

Reply:
yay nice

Reply:
jk it failed when i rebased rip